test/box-luatest/gh_4691_data/fill.lua

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env tarantool
+
+local fio = require('fio')
+
+-- {{{ Helpers to determine the script directory
+
+local function sourcefile()
+    local info = debug.getinfo(3, 'S')
+    local source = info and info.source
+    return source and source:startswith('@') and source:sub(2) or nil
+end
+
+local function sourcedir()
+    local source = sourcefile()
+    return fio.abspath(source and source:match('(.*)/') or '.')
+end
+
+-- }}} Helpers to determine the script directory
+
+local destdir = sourcedir()
+
+box.cfg({
+    -- Place the resulting snapshot near to this fill.lua script.
+    work_dir = destdir,
+    -- We need only the last snapshot.
+    checkpoint_count = 1,
+    -- Our goal is to generate a snapshot and we don't need WALs.
+    wal_mode = 'none',
+})
+
+-- Allow the testing code to connect to the instance.
+box.schema.user.grant('guest', 'super')
+
+-- Ensure that the snapshot is generated with certain schema
+-- version.
+--
+-- The gh-4691 test needs the schema of the 2.1.3 version.
+local schema_version_tuple = box.space._schema:get({'version'})
+assert(schema_version_tuple ~= nil)
+assert(schema_version_tuple[1] == 'version')
+assert(schema_version_tuple[2] == 2)
+assert(schema_version_tuple[3] == 1)
+assert(schema_version_tuple[4] == 3)
```

Comment:
Somewhy I thought the idea is to checkout target (in this case - 2.1.3) source code, build it, make the snap, and use it as is. Not try to simulate the old schema on master tarantool. But if it covers the case, then alright.

Btw, you will still have `_vcollation` here, right? Which means netbox would connect fine regardless of the schema version and the original bug is not covered then. Or is it?

Reply:
It doesn't try to simulate the old schema on the new master. Those assertions ensure that the script is run using an older version of tarantool.

There is no `_vcollation` on given schema version. So it is verified transitively.

Regarding automatic checkout+build: it is a bit fragile. We need to backport [two patches](https://github.com/tarantool/tarantool/wiki/User:Totktonada#build-old-tarantool-version) (see 2.5.1 instruction) to build tarantool 2.1.3 on a modern OS. In a future the set of the patches may be larger. Moreover, a build command may vary from OS to OS. This is not what I would like to automate.

However this work is done in [tt install tarantool](https://github.com/tarantool/tt/issues/91). If we'll want to automate it I would reuse an existing solution.

Reply:
I wasn't attentive enough, somewhy thought these lines were setting the schema, not checking it.